spec/java/util/StringBuilder.main.lsl

```diff
@@ -0,0 +1,1022 @@
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.setCharAt (@target self: StringBuilder, index: int, ch: char): void
+    {
+        _checkIndex(index);
+        val newStr: array<char> = action ARRAY_NEW("char", this.length);
+        var i: int = 0;
+        action LOOP_FOR(i, 0, this.length, +1, _setNewLength_loop(i, index, newStr, ch));
+        this.storage = action OBJECT_TO_STRING(newStr);
+    }
+
+
+    @Phantom proc _setNewLength_loop (i: int, index: int, newStr: array<char>, ch: char): void
+    {
+        if(i != index)
+            newStr[i] = action CALL_METHOD(this.storage, "charAt", [i]);
+        else
+            newStr[i] = ch;
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.trimToSize (@target self: StringBuilder): void
+    {
+
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.substring (@target self: StringBuilder, start: int): String
+    {
+        result = _substring(start, this.length);
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.substring (@target self: StringBuilder, start: int, end: int): String
+    {
+        result = _substring(start, end);
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.subSequence (@target self: StringBuilder, start: int, end: int): CharSequence
+    {
+        _checkRangeSIOOBE(start, this.length, this.length);
+        val sizeNewString: int = end - start;
+        val newStr: array<char> = action ARRAY_NEW("char", sizeNewString);
+
+        var i: int = 0;
+        action LOOP_FOR(i, start, end, +1, _newSubString_loop(i, newStr));
+        result = action OBJECT_TO_STRING(newStr);
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.getChars (@target self: StringBuilder, srcBegin: int, srcEnd: int, dst: array<char>, dstBegin: int): void
+    {
+        _checkRangeSIOOBE(srcBegin, srcEnd, this.length);
+        var n: int = srcEnd - srcBegin;
+        val dstLength: int = action ARRAY_SIZE(dst);
+        _checkRange(dstBegin, dstBegin + n, dstLength);
+
+        var i: int = 0;
+        action LOOP_FOR(i, srcBegin, srcEnd, +1, _getChars_loop(i, dstBegin, dst));
+    }
+
+
+    @Phantom proc _getChars_loop(i: int, dstBegin: int, dst: array<char>): void
+    {
+        dst[dstBegin] = action CALL_METHOD(this.storage, "charAt", [i]);
+        dstBegin += 1;
+    }
+
+
+    // within java.lang.AbstractStringBuilder
+    fun *.codePointCount (@target self: StringBuilder, beginIndex: int, endIndex: int): int
+    {
+        if (beginIndex < 0 || endIndex > this.length || beginIndex > endIndex)
+            action THROW_NEW("java.lang.IndexOutOfBoundsException", []);
+
+        val count: int = endIndex - beginIndex;
+        val newStr: array<char> = action ARRAY_NEW("char", this.length);
+        var i: int = 0;
+        action LOOP_FOR(i, 0, this.length, +1, _newSubString_loop(i, newStr));
+
+        result = action DEBUG_DO("Character.codePointCount(newStr, beginIndex, count)");
```

Comment:
This is too complex to me.
Why we can't just return an integer between [`(E - B)`...`(E - B)*2`] using `action SYBOLIC("int")`?
*(`2` - is a maximum number of code points in the largest character possible)*